crypto/keys/bip39/wordcodec.go

```diff
@@ -0,0 +1,62 @@
+package bip39
+
+import (
+	"strings"
+
+	"github.com/bartekn/go-bip39"
+)
+
+// ValidSentenceLen defines the mnemonic sentence lengths supported by this BIP 39 library.
+type ValidSentenceLen uint8
+
+const (
+	// FundRaiser is the sentence length used during the cosmos fundraiser (12 words).
+	FundRaiser ValidSentenceLen = 12
+	// FreshKey is the sentence length used for newly created keys (24 words).
+	FreshKey ValidSentenceLen = 24
+)
+
+// NewMnemonic will return a string consisting of the mnemonic words for
+// the given sentence length.
+func NewMnemonic(len ValidSentenceLen) (words []string, err error) {
+	// len = (ENT + checksum) / 11
+	var ENT int
```

Comment:
Consider renaming ENT to entropy_size. I would actually prefer this to be an expression and not a switch case though. I.e. `entropy_size = len* 11 - checksum`, and have checksum be a constant somewhere. (As right now checksum is a constant that just isn't written down)

Reply:
* This shouldn't block this PR

Reply:
Renamed to `entropySize`, cc @Liamsi on the second question

Reply:
Made an issue for this: #1410, we can merge this pr and fix that later.

Reply:
I named it `ENT` so that anyone comparing it to the spec can quickly see what it is:
https://github.com/bitcoin/bips/blob/master/bip-0039.mediawiki#generating-the-mnemonic
But `entropySize` might be better for anyone not comparing to the spec 👍 

Regarding the 2nd question: checksum isn't a constant (in the spec it can be any integer 4, 5, 6, 7, 8). We picked 4 for the fundraiser and 8 for new keys.

Reply:
Thanks for explaining the situation! These checksums are constants per fundraiser type them and they should be at the top of the file. (The switch statement for deriving it still makes sense then) Right now its not clear without running through the math what the checksums are.